node/src/accountant/db_access_objects/failed_payable_dao.rs

```diff
@@ -0,0 +1,778 @@
+        assert_eq!(result.get(&absent_hash), None);
+        assert_eq!(result.get(&another_present_hash), Some(&2u64));
+    }
+
+    #[test]
+    fn failure_reason_from_str_works() {
+        assert_eq!(
+            FailureReason::from_str("PendingTooLong"),
+            Ok(PendingTooLong)
+        );
+        assert_eq!(FailureReason::from_str("NonceIssue"), Ok(NonceIssue));
+        assert_eq!(
+            FailureReason::from_str("InvalidReason"),
+            Err("Invalid FailureReason: InvalidReason".to_string())
+        );
+    }
+
+    #[test]
+    fn retrieve_condition_display_works() {
+        assert_eq!(
+            FailureRetrieveCondition::ByReason(PendingTooLong).to_string(),
+            "WHERE reason IS 'PendingTooLong'"
+        );
+    }
+
+    #[test]
+    fn can_retrieve_all_txs() {
+        let home_dir =
+            ensure_node_home_directory_exists("failed_payable_dao", "can_retrieve_all_txs");
+        let wrapped_conn = DbInitializerReal::default()
+            .initialize(&home_dir, DbInitializationConfig::test_default())
+            .unwrap();
+        let subject = FailedPayableDaoReal::new(wrapped_conn);
+        let tx1 = FailedTxBuilder::default().hash(make_tx_hash(1)).build();
+        let tx2 = FailedTxBuilder::default()
+            .hash(make_tx_hash(2))
+            .nonce(1)
+            .build();
+        let tx3 = FailedTxBuilder::default().hash(make_tx_hash(3)).build();
+        subject
+            .insert_new_records(&vec![tx1.clone(), tx2.clone()])
+            .unwrap();
+        subject.insert_new_records(&vec![tx3.clone()]).unwrap();
+
+        let result = subject.retrieve_txs(None);
+
+        assert_eq!(result, vec![tx1, tx2, tx3]);
+    }
+
+    #[test]
+    fn can_retrieve_pending_too_long_txs() {
+        let home_dir =
+            ensure_node_home_directory_exists("sent_payable_dao", "can_retrieve_pending_txs");
+        let wrapped_conn = DbInitializerReal::default()
+            .initialize(&home_dir, DbInitializationConfig::test_default())
+            .unwrap();
+        let subject = FailedPayableDaoReal::new(wrapped_conn);
+        let tx1 = FailedTxBuilder::default()
+            .hash(make_tx_hash(1))
+            .reason(FailureReason::PendingTooLong)
+            .checked(0)
+            .build();
+        let tx2 = FailedTxBuilder::default()
+            .hash(make_tx_hash(2))
+            .reason(FailureReason::PendingTooLong)
+            .checked(1)
+            .build();
+        let tx3 = FailedTxBuilder::default()
+            .hash(make_tx_hash(3))
+            .reason(FailureReason::NonceIssue)
+            .build();
+        subject
+            .insert_new_records(&vec![tx1.clone(), tx2.clone(), tx3.clone()])
+            .unwrap();
+
+        let result = subject.retrieve_txs(Some(ByReason(PendingTooLong)));
+
+        assert_eq!(result, vec![tx1, tx2]);
+    }
+
+    #[test]
+    fn update_tx_failures_works() {
```

Comment:
I still don't know what use case you had in your head when you put up this method.

Repeating myself, we will probably ever need just to update the checked column, nothing else should change once inserted, except deletions, those should also be possible.